presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -68,18 +68,24 @@ public Analysis analyze(Statement statement, boolean isDescribe)
         return analysis;
     }
 
-    static void verifyNoAggregatesOrWindowFunctions(Metadata metadata, Expression predicate, String clause)
+    static void verifyNoAggregateWindowOrGroupingFunctions(Metadata metadata, Expression predicate, String clause)
     {
         AggregateExtractor extractor = new AggregateExtractor(metadata);
         extractor.process(predicate, null);
 
         WindowFunctionExtractor windowExtractor = new WindowFunctionExtractor();
         windowExtractor.process(predicate, null);
 
-        List<FunctionCall> found = ImmutableList.copyOf(Iterables.concat(extractor.getAggregates(), windowExtractor.getWindowFunctions()));
+        GroupingOperationAnalyzer.Extractor groupingOperationExtractor = new GroupingOperationAnalyzer.Extractor();
```

Comment:
Good idea but I'll leave this as follow on work unless @martint thinks this should be done as part of this pull request.